experimental/MatroidRealizationSpaces/src/SelfProjectingMatroids.jl

```diff
@@ -0,0 +1,447 @@
+          push!(ineqs,R(candidates[i]))
+        end
+      end
+    end
+    return ineqs[2:length(ineqs)]
+end
+
+
+@doc raw"""
+    selfprojecting_realization_matrix(m::Matroid, Bas::Vector{Int}, F::Ring;Ideal::Union{Ideal,Nothing} = nothing, check::Bool = true)
+
+    Function to compute a template for a selfprojecting realization. Returns the ambient ring and the matrix. The input Bas defines the basis of the matriod that is chosen to be represented by the identity matrix in the realization. 
+    The defining_ideal of the selfprojecting realization space of the given matroid can optionally be entered. Otherwise it will be computed. 
+    The kwarg check states whether the matroid will be checked for self-projectivity. The default is true.
+# Examples
+```jldoctest
+julia> m = matroid_from_nonbases([[1,2,3],[4,5,6]],6)
+Matroid of rank 3 on 6 elements
+
+julia> R, M =selfprojecting_realization_matrix(m,[1,2,4])
+┌ Warning: This function is slow except for small matroids!
+└ @ Oscar ~/Documents/Oscar.jl/experimental/MatroidRealizationSpaces/src/SelfProjectingMatroids.jl:198
+(Multivariate polynomial ring in 2 variables over QQ, [1 0 1 0 1 1; 0 1 1 0 x1 x1; 0 0 0 1 1 x2])
+
+julia> M
+[1   0   1   0    1    1]
+[0   1   1   0   x1   x1]
+[0   0   0   1    1   x2]
+
+
+julia> m = uniform_matroid(3,6)
+Matroid of rank 3 on 6 elements
+
+julia> R, M =selfprojecting_realization_matrix(m,[1,2,3])
+┌ Warning: This function is slow except for small matroids!
+└ @ Oscar ~/Documents/Oscar.jl/experimental/MatroidRealizationSpaces/src/SelfProjectingMatroids.jl:198
+(Quotient of multivariate polynomial ring by ideal (x1*x2*x3 - x1*x2*x4 - x1*x3*x4 + x1*x4 + x2*x3*x4 - x2*x3), [1 0 0 1 1 1; 0 1 0 1 x1 x3; 0 0 1 1 x2 x4])
+  ```
+"""
+###################
+#Bas are the given columns that will be the identity matrix
+#This function returns the ambient ring and the realization matrix, same as for the classical function realization_space_matrix
+function selfprojecting_realization_matrix(m::Matroid, Bas::Vector{Int};I::Union{Ideal,Nothing} = nothing, check::Bool = true)
+  if check 
+    @req is_selfprojecting(m) "The given matroid is not self projecting" 
+  end
+  if !(is_realizable(m,char=0))
+    return (nothing, nothing)
+  else
+  X = realization_matrix(realization_space(m,B=Bas,ground_ring=QQ)) #this matrix is not yet simplified by the defining ideal of the selfprojecting realization ideal
+  if isnothing(I) #this way the ideal needs only be computed once!
+    I = selfprojecting_realization_ideal(m);
+  else 
+    I = I
+  end
+  if is_zero(I)
+    return (base_ring(X),X) 
+  elseif is_one(I) #this means the matrix is not realizable by self-projecting points
+    return (base_ring(I), nothing)
+  else
+  R = base_ring(X)
+  QR, phi = quo(R,I)
+  return (QR, phi.(X))
+  end
+  end
+end
+
+
+@doc raw"""
+    selfprojecting_realization_space(m::Matroid;
+  B::Union{GroundsetType,Nothing}=nothing; check::Bool = true)
```

Comment:
this will fit on one line in the documentation which will be easier to read
```suggestion
    selfprojecting_realization_space(m::Matroid;  B::Union{GroundsetType,Nothing}=nothing; check::Bool = true)
```